package/gluon-mesh-vpn-core/src/respondd.c

```diff
@@ -0,0 +1,103 @@
+/*
+  Copyright (c) 2021, Aiyion <gluon@aiyionpri.me>
+  Copyright (c) 2016, Matthias Schiffer <mschiffer@universe-factory.net>
+  All rights reserved.
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <respondd.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#include <json-c/json.h>
+#include <libgluonutil.h>
+
+
+char * read_stdout(const char *command) {
+	FILE *f = popen(command, "r");
+	if (!f)
+		return NULL;
+
+	char *line = NULL;
+	size_t len = 0;
+
+	ssize_t r = getline(&line, &len, f);
+
+	pclose(f);
+
+	if (r >= 0) {
+		// The len given by getline is the buffer size, not the string length
+		len = strlen(line);
+
+		if (len && line[len-1] == '\n')
+			line[len-1] = 0;
+	} else {
+		free(line);
+		return NULL;
+	}
```

Comment:
I was a bit confused of the desired outcome for 1-char strings only containing `\n`, however as the else block is only for error handling, how about this:

```      
ssize_t r = getline(&line, &len, f);

pclose(f);
if (r < 0) {
	free(line);
	return NULL;
}

/* The len given by getline is the buffer size, not the string length */
len = strlen(line);

if (len && line[len-1] == '\n')
	line[len-1] = 0;

return line;
```

Note also the change from CPP comment style to C.

Reply:
implemented